blockpoller/state_file.go

```diff
@@ -33,20 +33,36 @@ type stateFile struct {
 	Blocks         []blockRefWithPrev
 }
 
+func (p *BlockPoller) isStateFileExist(stateStorePath string) bool {
+	if stateStorePath == "" {
+		p.logger.Info("No state store path set, skipping cursor check")
+		return false
+	}
+	fp := filepath.Join(stateStorePath, "cursor.json")
+	_, err := os.Stat(fp)
+	exist := err == nil || os.IsExist(err)
```

Comment:
I think `IsExists(err)` could be true only on doing a write, but `Stat` does a read only. It should be only `err == nil`